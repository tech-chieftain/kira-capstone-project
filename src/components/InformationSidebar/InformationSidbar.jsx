import PropTypes from "prop-types";
import Container from "./InformationSidebar.styled";

const InformationSidebar = ({ skills, languages, educations }) => (
  <Container className="shadow">
    <div className="w-75 mx-auto my-5 ">
      <h5 className="px-3 py-2 border-bottom border-light">Skills</h5>
      <div>
        <ul>
          {skills.map((skill, index) => (
            <li key={index}>{skill}</li>
          ))}
        </ul>
      </div>
    </div>

    <div className="w-75 mx-auto my-5">
      <h5 className="px-3 py-2 border-bottom border-light">Language</h5>
      <div>
        <ul>
          {languages.map((lang, index) => (
            <li key={index}>{lang}</li>
          ))}
        </ul>
      </div>
    </div>

    <div className="w-75 mx-auto my-5">
      <h5 className="px-3 py-2 border-bottom border-light">Education</h5>
      <div>
        <ul>
          {educations.map((education, index) => (
            <li key={index}>{education}</li>
          ))}
        </ul>
      </div>
    </div>
  </Container>
);

InformationSidebar.propTypes = {
  skills: PropTypes.array,
  languages: PropTypes.array,
  educations: PropTypes.array,
};

InformationSidebar.defaultProps = {
  skills: ["Html", "Javascript", "Css"],
  languages: ["Kurdish", "Arabic", "English"],
  educations: ["Salahaddin University", "College of Engineering"],
};
export default InformationSidebar;
