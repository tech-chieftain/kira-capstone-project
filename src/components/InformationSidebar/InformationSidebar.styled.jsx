import styled from "styled-components";

const Container = styled.div`
  width: 347px;
  height: 100%;
  padding: 10px 0 40px;
  background: white;
  margin: -2rem 0 0 0;

  ul li {
    margin: 5px 0;
  }

  #progress {
    height: 11px;
    border-radius: 20px;
    width: 194px;
    margin: 5px 0;
  }
`;

export default Container;
