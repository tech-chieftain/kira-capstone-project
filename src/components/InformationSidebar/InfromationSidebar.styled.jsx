import styled from "styled-components";

export const Container = styled.div`
  width: 347px;
  height: 100%;
  padding: 60px 30px;

  ul li {
    margin-bottom: 3px;
  }

  small {
    margin-right: 10px;
  }
`;

export const ProgressBar = styled.progress`
  margin-bottom: -5px;

    &[value] {
        -webkit-appearance: none;
        appearance: none;
    }

    &[value]::-webkit-progress-bar {
        height: 10px;
        border-radius: 20px;
        background: #eee;
        width: 194px;8
    }

    &[value]::-webkit-progress-value {
        height: 10px;
        border-radius: 20px;
        background: #719e9e;
    }
`;
