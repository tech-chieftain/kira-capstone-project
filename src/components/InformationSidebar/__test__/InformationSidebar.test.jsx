import React from "react";
import renderer from "react-test-renderer";
import InformationSidebar from "../InformationSidbar";

test("properly renders the InformationSidebar Component", () => {
  const tree = renderer.create(<InformationSidebar />).toJSON();
  expect(tree).toMatchSnapshot();
});
