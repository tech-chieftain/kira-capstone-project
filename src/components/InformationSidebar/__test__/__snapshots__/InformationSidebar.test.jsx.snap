// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly renders the InformationSidebar Component 1`] = `
<div
  className="InformationSidebarstyled__Container-sc-1rywt34-0 blvETZ shadow"
>
  <div
    className="w-75 mx-auto my-xl-5 my-md-0"
  >
    <h5
      className="px-3 py-2 border-bottom border-light"
    >
      profile.skills
    </h5>
    <div>
      <ul>
        <li>
          Html
        </li>
        <li>
          Javascript
        </li>
        <li>
          Css
        </li>
      </ul>
    </div>
  </div>
  <div
    className="w-75 mx-auto my-xl-5 my-md-0"
  >
    <h5
      className="px-3 py-2 border-bottom border-light"
    >
      profile.language
    </h5>
    <div>
      <ul>
        <li>
          Kurdish
        </li>
        <li>
          Arabic
        </li>
        <li>
          English
        </li>
      </ul>
    </div>
  </div>
  <div
    className="w-75 mx-auto my-xl-5 my-md-0"
  >
    <h5
      className="px-3 py-2 border-bottom border-light"
    >
      profile.education
    </h5>
    <div>
      <ul>
        <li>
          Salahaddin University
        </li>
        <li>
          College of Engineering
        </li>
      </ul>
    </div>
  </div>
</div>
`;
